streamingChallengeJuvenal/src/main/scala/TwitterHbcClient.scala

```diff
@@ -0,0 +1,69 @@
+import com.twitter.hbc.httpclient.auth._
+import com.twitter.hbc.core.{Client, Constants}
+import com.twitter.hbc.core.endpoint.{StatusesFilterEndpoint, Location}
+import java.util.concurrent.{BlockingQueue, LinkedBlockingQueue, TimeUnit}
+import com.twitter.hbc.ClientBuilder
+import com.twitter.hbc.core.processor.StringDelimitedProcessor
+
+import scala.collection.JavaConverters._
+import org.apache.log4j.{LogManager, Logger}
+import KafkaProducerClient._
+
+
+// I used twitter Hosebird Client API to develop a specific client that will takes data from a specific place, and stored them in a Kafka topic
+object TwitterHbcClient {
+
+  private var trace_client_streaming: Logger = LogManager.getLogger("Log_Console")
+
+
+  /**
+   * The Hosebird client
+   *
+   * @param CONSUMER_KEY          : the OAuth consumer key of the twitter account
+   * @param CONSUMER_SECRET       : the OAuth consumer secret of the twitter account
+   * @param ACCESS_TOKEN          : the OAuth access token of the twitter account
+   * @param TOKEN_SECRET          : the OAuth token secret  of the twitter account
+   * @param town_list             : the specific place from where we collect the tweets
+   * @param kafkaBootStrapServers : Kafka brokers IP adresses
+   * @param topic                 : topic to use in Kafka
+   */
+
+  def ProducerTwitterKafkaHbc(CONSUMER_KEY: String, CONSUMER_SECRET: String, ACCESS_TOKEN: String, TOKEN_SECRET: String, town_list: Location,
+                              kafkaBootStrapServers: String, topic: String): Unit = {
+
+    val queue: BlockingQueue[String] = new LinkedBlockingQueue[String](100000)
+
```

Comment:
Hello Juvénal ,thanks for your PR and for taking the time to do the technical challenge 
I have some questions here ...

why a `LinkedBlockingQueue`  ? 
what are the performance implications being bounded to `100000` ?  
how would you scale the client using a bounded queue ? any impact to the consumers ? 
What would be the issues if is unbounded ? 
any alternative to this queue  if there is such ?

Reply:
Hello @sajid-austria-byd , you are welcome ! 

About your questions, I used a LinkedBlockingQueue mainly because I wanted to avoid a heap memory error due to the vast potential amount of tweets that the HBC client can receive. 
Also, a queue will allow me to order the tweets so that they will end up being ordered in Kafka. Without the order, we might end up with incorrect KPI, since they are time-based.

Yes, the bound to 100000 is just arbritary, because the  actual size of the queue will depend on the PC's memory where the HBC client will be deployed. The implication is that, the higher the twitter's throughput, the larger the queue  (and the memory of the PC) will need to be. 

To scale the client, we will have to increase the bound of the queue depending on the throughput the client receive and also increase the VM/PC's memory. The queue is not the ultimate problem here for scalability. It is the memory of client that is the issue. using a bounded queue allow us to control the thoughtput and to avoid the client to fall of itself because of a heat memory error. 

No, there is no impact on the consumers since the data are collected  in Kafka and persisted on the disk. This scalability issued is solely related to the hbc client.  

Yeah, I also though briefly about using an unbounded queue, but by doing so, we will not have any control over the throughput managed by the client. So, the client will be more prompted to heap memory. Without a bound, the queue will just follow automatically the memory allocated to its by the pc, until its falls. But, with a bounded queue, we know for sure that the throughput will be limited to the bound that we set, and will not try to go over that. 

Hope my answers clarify the choices I made and answered your questions well. If not, do not hesitate to ask other questions. 

Thanks, 
Juvenal